src/passes/OptimizeCasts.cpp

```diff
@@ -102,6 +120,273 @@ namespace wasm {
 
 namespace {
 
+// Contains information about a RefCast we want to move to some target.
+struct RefCastInfo {
+  Type type;
+  RefCast::Safety safety;
+
+  // Indicate whether the cast is already at the target. If it is, then
+  // the information is just for determining if a potential cast is less
+  // refined than the existing cast, and will be removed later.
+  bool atTarget;
+
+  RefCastInfo() {}
+
+  RefCastInfo(Type type, RefCast::Safety safety, bool atTarget)
+    : type(type), safety(safety), atTarget(atTarget) {}
+
+  void modify(Type newType, RefCast::Safety newSafety, bool newAtTarget) {
+    type = newType;
+    safety = newSafety;
+    atTarget = newAtTarget;
+  }
```

Comment:
I don't think this is needed, since you can replace `foo.modify(x, y, z)` with `foo = RefCastInfo(x, y, z)`.

Reply:
Removed.